java/flight/flight-core/src/main/java/org/apache/arrow/flight/RenewFlightEndpointRequest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.flight;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.ByteBuffer;
+import java.util.Objects;
+
+import org.apache.arrow.flight.impl.Flight;
+
+/** A request to extend the expiration time of a FlightEndpoint. */
+public class RenewFlightEndpointRequest {
+  private final FlightEndpoint endpoint;
+
+  public RenewFlightEndpointRequest(FlightEndpoint endpoint) {
+    this.endpoint = Objects.requireNonNull(endpoint);
+  }
+
+  RenewFlightEndpointRequest(Flight.RenewFlightEndpointRequest proto) throws URISyntaxException {
+    this(new FlightEndpoint(proto.getEndpoint()));
+  }
+
+  public FlightEndpoint getFlightEndpoint() {
```

Comment:
@lidavidm How about renaming this to `GetEndpoitn()` because `FlightInfo` uses `getEndpoints()` not `getFlightEndpoints()`?